space/authz.go

```diff
@@ -0,0 +1,116 @@
+type AuthzServiceManager interface {
+	AuthzService() AuthzService
+	EntitlementEndpoint() string
+}
+
+// KeyclaokAuthzServiceManager is a keyaloak implementation of a space autharizarion service
+type KeyclaokAuthzServiceManager struct {
+	Service             AuthzService
+	entitlementEndpoint string
+}
+
+// AuthzService returns a space autharizarion service
+func (m *KeyclaokAuthzServiceManager) AuthzService() AuthzService {
+	return m.Service
+}
+
+// EntitlementEndpoint returns a keyclaok entitlement endpoint URL
+func (m *KeyclaokAuthzServiceManager) EntitlementEndpoint() string {
+	return m.entitlementEndpoint
+}
+
+// KeyclaokAuthzService implements AuthzService interface
+type KeyclaokAuthzService struct {
+	config AuthzConfiguration
+}
+
+// NewAuthzService constructs a new KeyclaokAuthzService
+func NewAuthzService(config AuthzConfiguration) *KeyclaokAuthzService {
+	return &KeyclaokAuthzService{config: config}
+}
+
+// Authorize returns true and the corresponding Requesting Party Token if the current user is among the space collaborators
+func (s *KeyclaokAuthzService) Authorize(ctx context.Context, entitlementEndpoint string, spaceID string) (*string, bool, error) {
+	token := goajwt.ContextJWT(ctx)
+	if token == nil {
+		return nil, false, errs.NewUnauthorizedError("missing token")
+	}
+
+	resource := auth.EntitlementResource{
+		Permissions: []auth.ResourceSet{{Name: spaceID}},
+	}
+	ent, err := auth.GetEntitlement(ctx, entitlementEndpoint, resource, token.Raw)
+	if err != nil {
+		return nil, false, err
+	}
+	return nil, ent != nil, nil
+}
+
+// Configuration returns authz service configuration
+func (s *KeyclaokAuthzService) Configuration() AuthzConfiguration {
+	return s.config
+}
+
+// InjectAuthzService is a middleware responsible for setting up AuthzService in the context for every request.
+func InjectAuthzService(service AuthzService) goa.Middleware {
+	return func(h goa.Handler) goa.Handler {
+		return func(ctx contx.Context, rw http.ResponseWriter, req *http.Request) error {
+			config := service.Configuration()
+			var endpoint string
+			if config != nil {
+				var err error
+				endpoint, err = config.GetKeycloakEndpointEntitlement(&goa.RequestData{Request: req})
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{
+						"err": err,
+					}, "unable to get entitlement endpoint")
+					return err
+				}
+			}
+			ctxWithAuthzServ := tokencontext.ContextWithSpaceAuthzService(ctx, &KeyclaokAuthzServiceManager{Service: service, entitlementEndpoint: endpoint})
+			return h(ctxWithAuthzServ, rw, req)
+		}
+	}
+}
+
+// Authorize returns true and the corresponding Requesting Party Token if the current user is among the space collaborators
+func Authorize(ctx context.Context, spaceID string) (*string, bool, error) {
+	srv := tokencontext.ReadSpaceAuthzServiceFromContext(ctx)
+	if srv == nil {
+		log.Error(ctx, map[string]interface{}{
+			"authz-service": srv,
```

Comment:
Wait a minute. I started to avoid capitalized error messages we through to users. I have to if I want to get your approval on my PRs :) Which makes sense because the error messages may be concatenated.
But it doesn't matter if we use capitalized messages in logs, does it? :) It's just:
`err: "Something wrong"`
vs.
`err: "something wrong"`
in our logs, right?